packages/r/rsm-mmio/xmake.lua

```diff
@@ -0,0 +1,30 @@
+package("rsm-mmio")
+    set_homepage("https://github.com/Ryan-rsm-McKenzie/mmio")
+    set_description("A cross-platform memory-mapped io library for C++")
+    set_license("MIT")
+
+    add_urls("https://github.com/Ryan-rsm-McKenzie/mmio/archive/refs/tags/$(version).tar.gz",
+             "https://github.com/Ryan-rsm-McKenzie/mmio.git")
+    add_versions("2.0.0", "360dddf74a97bd0a7eb41378cc59f2a69871dabfd36c55bf027429ac54930d5b")
+
+    on_install(function (package)
+        io.writefile("xmake.lua", [[
+            add_rules("mode.debug", "mode.release")
+            set_languages("c++17")
+            target("rsm-mmio")
+                set_kind("static")
+                add_files("src/**.cpp")
+                add_includedirs("include/")
+                add_headerfiles("include/(**.hpp)")
+        ]]) 
```

Comment:
we need support static/shared. `configs.kind = "shared"`, you can see other packages. e.g. zlib/libpng